### PR TITLE
Add an example of option pricing with multiple GPUs

### PR DESCRIPTION
**Pull request by rezoo**
Merge #473 and #493 first.
This PR add an example that computes the price of call options using Monte-Carlo simulation and multiple GPUs.

### CODE REVIEW

**Comment by okuta** [comment]
Please rebase or merge master.

**Comment by rezoo** [comment]
I rebased and add the test code. Please check it out.

File: tests/example_tests/test_finance.py [hunk 1]
```diff
@@ -34,3 +34,21 @@ def test_monte_carlo(self):
             r'    # of samples per option: 100\n' +
             r'GPU \(CuPy, Monte Carlo method\):\t[0-9\.]+ sec\n' +
             r'Error: [0-9\.]+')
+
+
+class TestMonteCarloWithMultiGPU(unittest.TestCase):
+
+    def test_monte_carlo(self):
+        output = example_test.run_example(
```
Comment (okuta) [inline tests/example_tests/test_finance.py]: Please use multi GPU.

**Comment by rezoo** [comment]
I fixed.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**